### PR TITLE
🔐 Auth state integration

### LINKED ISSUE #190: Authentication state & integration
## Description

We need to prepare our app to handle authentication.

This involves:
- Creating "async thunk action creators" in Redux for logging in and logging out. These thunks get the token from the API
- When logging in and out, persist/remove the token from a persistent form of storage, in addition to the Redux store.
  - There will be a "Remember Me" checkbox on the login page.
  - Depending on whether this is checked, what has to be different about how you store the token?

Note that this issue does not include the UI for logging in/out.

## Learning objectives

- Researching and designing the best possible approach to integrate an app-wide construct into its architecture
- How authentication works with JWT
- Using Redux as a state management tool for information that concerns the entire app
- `localStorage` and `sessionStorage`

## Todo

- [x] Read through all the links below
- [x] Re-read this issue
- [x] Feel free to do additional reading (articles, examples) on how to approach designing this system
- [x] Create your own plan (a list of todos) and add them to the pull request in a reply for me to review
- [x] After I look over it, start implementing

## Useful links

- https://jwt.io/introduction
- https://redux.js.org/faq/miscellaneous#how-can-i-implement-authentication-in-redux
  - Please note that we're using a library called Redux Toolkit, which can handle many things for you.
  - Be sure to get an understanding of what Redux Toolkit provides (links below) before diving in.
- https://redux.js.org/tutorials/essentials/part-1-overview-concepts
- https://redux.js.org/tutorials/essentials/part-5-async-logic
  - In particular, note the `createAsyncThunk` API.
- Users API from https://petstore.swagger.io/?url=https://fallingfruit.org/test-api/0.3/openapi.yml
- https://redux-toolkit.js.org/usage/usage-guide

## Other links
- https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage
- https://dev.to/link2twenty/react-redux-and-localstorage-2lih
- https://redux-toolkit.js.org/usage/usage-guide#asynchronous-logic-and-data-fetching

### PR DESCRIPTION
**Pull request by daniel-moon32**
## Status:
:rocket: Ready

## Description
Add integration with auth token

Fixes #190  

## Screenshots
Screen Shot 2021-11-02 at 2 34 11 AM

### CODE REVIEW

File: src/redux/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+
+import { getUserToken } from '../utils/api'
+
+const authToken = 'authToken'
+
+export const fetchAccessToken = createAsyncThunk(
+  'users/fetchToken',
+  async ({ email, password, isChecked }) => {
```
Comment (Qrtn) [inline src/redux/authSlice.js:9]: change isChecked to `rememberMe`

File: src/components/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  }
+`
+
+const ErrorMessage = styled.p`
+  color: red;
+`
+const LoginPage = () => {
+  const { authToken, failedLogin } = useSelector((state) => state.auth)
+  const history = useHistory()
+  const [isChecked, setIsChecked] = useState(false)
```
Comment (Qrtn) [inline src/components/LoginPage.js:49]: rememberMe. isChecked is too vague

File: src/utils/AuthInitializer.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { useDispatch } from 'react-redux'
+
+import { login } from '../redux/authSlice'
+
+const AuthInitializer = () => {
+  const dispatch = useDispatch()
+  useEffect(() => {
+    dispatch(login())
+  }, [dispatch])
+  return <></>
```
Comment (Qrtn) [inline src/utils/AuthInitializer.js:11]: You can just return null here.

File: src/redux/authSlice.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+
+import { getUserToken } from '../utils/api'
+
+const authToken = 'authToken'
+
+export const fetchAccessToken = createAsyncThunk(
```
Comment (Qrtn) [inline src/redux/authSlice.js:7]: To be consistent, you should just rename to `fetchToken`.

**Review by Qrtn — APPROVED** [review]
Great job, guys! So excited to for y'all start working with this infrastructure as we integrate auth into the rest of the app.

File: src/components/desktop/Header.js [hunk 5]
```diff
@@ -201,9 +213,23 @@ const Header = () => {
...
           <li>
-            <button>
-              <User /> {t('Login')}
-            </button>
+            {authToken ? (
+              <StyledDropdown
+                label={
+                  <StyledUser>
+                    <User height={'15px'} />{' '}
+                    {authToken ? 'Arman Rafati' : t('Login')}
```
Comment (SirajChokshi) [inline src/components/desktop/Header.js:221]: Lets add a TODO here also, and @code-arman can use somebody else's name here (John Doe, etc) if you care

**Review by SirajChokshi — APPROVED** [review]
Few nits, but looks good overall 🚀

**Pull request merged**